server/aesdsocket.c

```diff
@@ -295,8 +299,31 @@ void* threadfunc(void* thread_param)
 
         //Writing to file
         int new_testfile_fd = open(filename, O_RDWR | O_CREAT | O_APPEND, S_IWUSR | S_IRUSR | S_IWGRP | S_IRGRP | S_IROTH);
-        int nr = write(new_testfile_fd, storage_buffer, bytes_to_write);
+
+        int nr = 0;
+
+        // Added case for IOCTL command
+        if (strncmp(storage_buffer, aesd_ioctl_str, strlen(aesd_ioctl_str)) == 0) {
+
+            struct aesd_seekto seekto;
+
+            // Using sscanf to divide string and store values for write command and write command offset
+            sscanf(storage_buffer, "AESDCHAR_IOCSEEKTO:%d,%d", &seekto.write_cmd, &seekto.write_cmd_offset);
```

Comment:
@GautamaGandhi missing return code and error handling here, which could mean you call ioctl with uninitialized arguments in `seekto`.  Would be best to zero out the structure before calling sscanf and also to check the return value to ensure the expected number of parameters were filled.

Reply:
@dwalkes yes, I shall fix this by adding a return statement to ensure that the correct parameters are filled.